src/marqo/marqo_cloud_instance_mappings.py

```diff
@@ -60,7 +61,9 @@ def _refresh_urls(self, timeout=None):
             mq_logger.warning(response.text)
         response_json = response.json()
         for index in response_json['results']:
-            if index.get('index_status') in ["READY", "CREATING"]:
-                self._urls_mapping[index['index_status']][index['index_name']] = index.get('endpoint')
+            if index.get('index_status') in [IndexStatus.READY, IndexStatus.MODIFYING]:
+                self._urls_mapping[IndexStatus.READY][index['index_name']] = index.get('endpoint')
+            elif index.get('index_status') == IndexStatus.CREATING:
```

Comment:
Please add state transition tests for this function. i.e. creating -> ready, ready -> deleting/deleted, ready->modifying, modifying->ready. In each test ensure mappings are left in a valid state. I suspect currently creating->ready won't clean up the record from Creating, which looking at the rest of the code won't break anything currently but is a risk and should be addressed.